sdk/src/client/api/block_builder/transaction_builder/requirement/account.rs

```diff
@@ -42,7 +36,13 @@ impl TransactionBuilder {
                 "{account_id:?} requirement already fulfilled by {:?}",
                 input.output_id()
             );
-            return Ok(Vec::new());
+            return Ok(());
+        }
+
+        if !self.allow_additional_input_selection {
```

Comment:
Can we not move this to `select_input`?
Even if it means passing it the current Requirement

Reply:
We can't, because we call select_inputs on required inputs too 😄

Reply:
![image](https://github.com/iotaledger/iota-sdk/assets/3305068/60b8b96a-96bb-49b2-b9e5-3ad2c88c416f)

Reply:
Honestly tho it's fine because the check is not a lot of duplicated code and it allows us to easily add the requirement too 🤷‍♀️